### PR TITLE
Adds image data to the default cube fixture and enlarge tests

### PR DESCRIPTION
**Pull request by kaitlyndlee**
## Description

Added image data to the default cube fixture for tests. 
Refactored enlarge to be callable.
Wrote unit tests for enlarge.

## Related Issue

Part of testing. No issue.

## Motivation and Context

Adds the ability to test with a cube that has image data.

## How Has This Been Tested?

Makefile tests passed after refactoring enlarge. Newly created tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/Fixtures.cpp [hunk 1]
```diff
@@ -21,6 +23,14 @@ namespace Isis {
     testCube = new Cube();
     testCube->fromIsd(tempDir.path() + "/default.cub", label, isd, "rw");
 
+    LineManager line(*testCube);
```
Comment (kaitlyndlee) [inline isis/tests/Fixtures.cpp:26]: Decided to use a LineManager because it was easier to follow along with. Could have used a Brick

**Comment by kaitlyndlee** [comment]
[enlarge test conversion](https://trello.com/c/ONgxgEng/108-enlarge-test-conversion)

**Comment by kaitlyndlee** [comment]
[Mocking / Faking image data](https://trello.com/c/U4sZ5IhE/206-mocking-image-data)

File: isis/src/base/apps/enlarge/enlarge_app.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#include "enlarge_app.h"
+#include "IException.h"
+#include "ProcessRubberSheet.h"
+
+using namespace std;
+using namespace Isis;
+namespace Isis{
+
+  void enlarge(UserInterface &ui, Pvl *log) {
+    Cube *cube = new Cube(ui.GetFileName("FROM"), "r");
```
Comment (jessemapel) [inline isis/src/base/apps/enlarge/enlarge_app.cpp]: You need to check for CubeAttributeInputs:
https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/apps/cam2map/cam2map.cpp#L29

File: isis/src/base/apps/enlarge/enlarge_app.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    CubeAttributeInput inAtt = ui.GetInputAttribute("FROM");
+    if (inAtt.bands().size() != 0) {
+      cube->setVirtualBands(inAtt.bands());
+    }
```
Comment (jessemapel) [inline isis/src/base/apps/enlarge/enlarge_app.cpp:17]: You have to set the virtual bands before you open the cube. You need to make a new Cube object without passing a filename, set the virtual bands, and then open the cube.

File: isis/tests/Fixtures.cpp [hunk 4]
```diff
@@ -21,6 +23,14 @@ namespace Isis {
...
+    for(line.begin(); !line.end(); line++) {
+      for(int i = 0; i < line.size(); i++) {
+        line[i] = (double) 1;
+      }
+      testCube->write(line);
+    }
```
Comment (Kelvinrr) [inline isis/tests/Fixtures.cpp:32]: Maybe I am missing something, but it seems like you are adding lines on the fly here, but the fixture on disk also has binary data?

Comment (jessemapel) [reply isis/tests/Fixtures.cpp:32]: The LineManager loops over the lines that are already in the cube.

Comment (Kelvinrr) [reply isis/tests/Fixtures.cpp:32]: sounds reasonable enough

**Review by Kelvinrr — APPROVED** [review]
Assuming Jesse's thinks his comment of adequately addressed this seems mergeable.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
Waiting for CI to finish

**Pull request merged**